iceoryx_posh/include/iceoryx_posh/internal/popo/modern_api/base_subscriber.inl

```diff
@@ -104,46 +107,69 @@ inline cxx::expected<cxx::optional<Sample<const T>>, ChunkReceiveError> BaseSubs
     }
 }
 
-template <typename T, typename port_t>
-inline void BaseSubscriber<T, port_t>::releaseQueuedSamples() noexcept
+template <typename T, typename Subscriber, typename port_t>
+inline void BaseSubscriber<T, Subscriber, port_t>::releaseQueuedSamples() noexcept
 {
     m_port.releaseQueuedChunks();
 }
 
-template <typename T, typename port_t>
-inline void
-BaseSubscriber<T, port_t>::setConditionVariable(ConditionVariableData* const conditionVariableDataPtr) noexcept
+template <typename T, typename Subscriber, typename port_t>
+inline void BaseSubscriber<T, Subscriber, port_t>::invalidateTrigger(const uint64_t uniqueTriggerId) noexcept
 {
-    m_port.setConditionVariable(conditionVariableDataPtr);
+    if (m_trigger.getUniqueId() == uniqueTriggerId)
+    {
+        m_port.unsetConditionVariable();
+        m_trigger.invalidate();
+    }
 }
 
-template <typename T, typename port_t>
-inline void BaseSubscriber<T, port_t>::unsetConditionVariable() noexcept
+// ============================== Sample Deleter ============================== //
+
+template <typename T, typename Subscriber, typename port_t>
+inline BaseSubscriber<T, Subscriber, port_t>::SubscriberSampleDeleter::SubscriberSampleDeleter(port_t& port)
+    : m_port(std::ref(port))
 {
-    m_port.unsetConditionVariable();
 }
 
-template <typename T, typename port_t>
-inline bool BaseSubscriber<T, port_t>::hasTriggered() const noexcept
+template <typename T, typename Subscriber, typename port_t>
+inline void BaseSubscriber<T, Subscriber, port_t>::SubscriberSampleDeleter::operator()(T* const ptr) const
 {
-    return m_port.hasNewChunks();
+    auto header = mepoo::convertPayloadPointerToChunkHeader(reinterpret_cast<void*>(ptr));
+    m_port.get().releaseChunk(header);
 }
 
-// ============================== Sample Deleter ============================== //
-
-template <typename T, typename port_t>
-inline BaseSubscriber<T, port_t>::SubscriberSampleDeleter::SubscriberSampleDeleter(port_t& port)
-    : m_port(std::ref(port))
+template <typename T, typename Subscriber, typename port_t>
+inline cxx::expected<WaitSetError>
+BaseSubscriber<T, Subscriber, port_t>::attachTo(WaitSet& waitset,
+                                                [[gnu::unused]] const SubscriberEvent subscriberEvent,
+                                                const uint64_t triggerId,
+                                                const Trigger::Callback<Subscriber> callback) noexcept
 {
+    using SelfType = BaseSubscriber<T, Subscriber, port_t>;
```

Comment:
When using this kind of template pattern, this could be useful to be defined in the header (as SelfType or ThisType).

Reply:
You are right. Fixed.